File: Resources/Projects/Archive/2023/Competition/Programming/Contest/TOI-Zero_68/A2/019/TOI-Zero_68/A2/024/024.py
n, k = map(int, input().split())
rb = mn = fr = 0
a, b, c = map(int, input().split())

for _ in range(k):
    x, y = map(int, input().split())
    if x % a == 0:
        rb += y
    if x % b == 0:
        mn += y
    if x % c == 0:
        fr += y

mx = max(rb, mn, fr)
if rb == mx:
    print("Rabbit", rb)
if mn == mx:
    print("Monkey", mn)
if fr == mx:
    print("Frog", fr) 

File: TOI-Zero_68/A2/024/024.cpp
#include<bits/stdc++.h>
using namespace std;

#define ll long long

int main(){
	ios::sync_with_stdio(0);
	cin.tie(0);
	int n,k;
    cin >> n >> k;
    int rb=0,mn=0,fr=0,a,b,c;
    cin >> a >> b >> c;
    while(k--){
        int x,y;
        cin >> x >> y;
        if(x%a==0){
        	rb += y;
		}
        if(x%b==0){
        	mn += y;
		}
        if(x%c==0){
        	fr += y;
		}
    }
    if(rb==max({rb,mn,fr})){
    	cout << "Rabbit " << rb << endl;
	}
    if(mn==max({rb, mn, fr})){
    	cout << "Monkey " << mn << endl;
	}
    if(fr==max({rb, mn, fr})){
    	cout << "Frog " << fr << endl;
	}
}
